lib/assessment.rb

```diff
@@ -43,6 +49,7 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    File.open(filename) do |file|
+      file.gsub(/"\n"/, ' ').each_with_object([]){|i, a| i << a}
```

Comment:
Very close! You should also trim that final space at the end of the last line - the method to do that is `strip` (or `rstrip`, since it's specfically trailing whitespace that we're talking about).